analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -141,13 +141,69 @@ class Bower(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfigu
             return result
         }
 
+        private fun hasCompleteDependencies(node: JsonNode, scopeName: String): Boolean {
+            val dependencyKeys = node["dependencies"].fieldNamesOrEmpty().asSequence().toSet()
+            val dependencyRefKeys = node["pkgMeta"][scopeName].fieldNamesOrEmpty().asSequence().toSet()
+
+            return dependencyKeys.containsAll(dependencyRefKeys)
+        }
+
+        private fun dependencyKeyOf(node: JsonNode): String? {
+            // As non-null dependency keys are supposed to as equivalence relation for parsing 'missing' nodes,
+            // thus only the name and version attributes can be used. Typically those attributes should be not null
+            // however in particular for root projects the null case also happens.
+            val name = node["pkgMeta"]["name"].textValueOrEmpty()
+            val version = node["pkgMeta"]["version"].textValueOrEmpty()
+            return if (name.isEmpty() || version.isEmpty()) null else "$name:$version"
+        }
+
+        private fun getNodesWithCompleteDependencies(node: JsonNode): Map<String, JsonNode> {
+            val result = mutableMapOf<String, JsonNode>()
+
+            val stack = Stack<JsonNode>()
+            stack.push(node)
+            while (!stack.empty()) {
+                val currentNode = stack.pop()
+
+                val key = dependencyKeyOf(currentNode)
+                if (key != null && hasCompleteDependencies(node, SCOPE_NAME_DEPENDENCIES)
+                        && hasCompleteDependencies(node, SCOPE_NAME_DEV_DEPENDENCIES)) {
+                    result[key] = currentNode
+                }
+
+                stack.addAll(getDependencyNodes(currentNode))
+            }
+
+            return result
+        }
+
         private fun extractDependencyTree(node: JsonNode, scopeName: String): SortedSet<PackageReference> {
+            return extractDependencyTree(node, scopeName, getNodesWithCompleteDependencies(node))
+        }
+
+        private fun extractDependencyTree(node: JsonNode, scopeName: String, alternativeNodes: Map<String, JsonNode>):
```

Comment:
Use a default argument here for `alternativeNodes` instead of adding the above wrapper function.